### ISSUE TITLE
Number input resets when typing period on Firefox 113

### ISSUE THREAD

**Issue by Tuoris** [solidjs/solid#1756]
### Describe the bug

Number input resets when typing period on Firefox 113.

Here is source code for minimal working example:
```
function App() {
    const [value, setValue] = createSignal("");

    return (
        <>
            <input type="number" value={value()} onInput={(event) => setValue(event.currentTarget.value)} />
            <div>Current value: "{value()}"</div>
        </>
    )
}
```

### Your Example Website or App

https://tuoris.neocities.org/solid-input-number-minimal-example/

### Steps to Reproduce the Bug or Issue

1. Run minimal working example or [open this page](https://tuoris.neocities.org/solid-input-number-minimal-example/)
2. Focus on input
3. Type: "4", ".", "2"

### Expected behavior

Input value should be "4.2"

### Screenshots or Videos

https://github.com/solidjs/solid/assets/16479460/0a285676-6475-4c9a-90b6-ae83bf350f76

### Platform

- Browser: Firefox
- Version: 113

### Additional context

I've [discussed this issue with Firefox team](https://bugzilla.mozilla.org/show_bug.cgi?id=1836659) - looks like they are following [W3C standard for valid floating point number](https://www.w3.org/TR/2011/WD-html5-20110525/common-microsyntaxes.html#valid-floating-point-number)

**Comment by Tuoris** [comment]
Similar example works in React (although onChange/onInput semantics is different from SolidJS), maybe these lines can be helpful to solve the issue - https://github.com/facebook/react/blob/main/packages/react-dom-bindings/src/client/ReactDOMInput.js#L118

**Comment by titoBouzout** [comment]
I can sympathize with the problem, but this is not a bug. The example code is doing `setValue(event.currentTarget.value)` which translates to `setValue("")` in firefox when the number ends with a ".". 
So Solidwill do just that and empty the value of the number, which seems correct to me. You probably want to do `event.currentTarget.value !== '' && setValue(event.currentTarget.value)`

**Comment by titoBouzout** [comment]
Didn't know of valueAsNumber. :) 

It is still an invalid issue because if you think about it, you are just using a signal. There's no way for the signal to know that's an input of the type number, or that the data comes from an event, nor really doesn't care because you just do `getValue` and `setValue` anywhere with any kind of data. 

I still sympathize, but the way I see it, if you start messing with events then you enter dark waters, you will end defining a whole new behaviour like the way  `onChange`, `onInput` works in react.

**Comment by atk** [comment]
My preferred way is not to overwrite the number on every input, but let the browser manage the field. Using `/*@once*/` or `untrack(value)`, you can set the value only once on instantiation, and then just read the value from the event (the browser should ensure you have the same value in your field as in your state).

**Comment by katywings** [comment]
Personally I also would love a standard approach for controlled number inputs handling these edgecases, as they also happened to me in the past. 

In the meantime @mattbdc what you maybe will find useful is something like this: 
https://playground.solidjs.com/anonymous/88e0b341-340d-4f32-94f3-a9b21f7890ee

I wrote the `numberInputHelper` by looking at your description and at the react way :).

**Comment by atk** [comment]
I really don't understand the obsession with controlled inputs, but if you must do it, you can use the `@solid-primitives/selection` primitive to fix the selection after overwriting the value.

**Comment by Tuoris** [comment]
Hi, @ryansolid !

The reason this issue happens in Firefox 114 because they recently decided to match a specs: [https://www.w3.org/TR/2011/WD-html5-20110525/common-microsyntaxes.html#valid-floating-point-number](https://www.w3.org/TR/2011/WD-html5-20110525/common-microsyntaxes.html#valid-floating-point-number). Chrome do not follow the specs in this case, and Safari - does. I've reported this behaviour as bug to Firefox team ([https://bugzilla.mozilla.org/show_bug.cgi?id=1836659](https://bugzilla.mozilla.org/show_bug.cgi?id=1836659)) - but they won't fix it too.

For my personal project I will stick with current workaround, but someone will probably benefit from **using text input instead of number input** (as GOV.UK did - https://technology.blog.gov.uk/2020/02/24/why-the-gov-uk-design-system-team-changed-the-input-type-for-numbers/). And also, if there was a way to get a both number and raw text value for number input in vanilla JS - that would be great.
